dask/dataframe/core.py

```diff
@@ -5428,25 +5431,31 @@ def cov_corr_agg(data, cols, min_periods=2, corr=False, scalar=False):
     return pd.DataFrame(mat, columns=cols, index=cols)
 
 
-def pd_split(df, p, random_state=None):
+def pd_split(df, p, random_state=None, shuffle=False):
     """ Split DataFrame into multiple pieces pseudorandomly
 
     >>> df = pd.DataFrame({'a': [1, 2, 3, 4, 5, 6],
     ...                    'b': [2, 3, 4, 5, 6, 7]})
 
-    >>> a, b = pd_split(df, [0.5, 0.5], random_state=123)  # roughly 50/50 split
+    >>> a, b = pd_split(
+    ...     df, [0.5, 0.5], random_state=123, shuffle=True
+    ... )  # roughly 50/50 split
     >>> a
        a  b
-    1  2  3
-    2  3  4
+    3  4  5
+    0  1  2
     5  6  7
     >>> b
        a  b
-    0  1  2
-    3  4  5
+    1  2  3
     4  5  6
+    2  3  4
     """
     p = list(p)
+    if shuffle:
+        if not isinstance(random_state, np.random.RandomState):
+            random_state = np.random.RandomState(random_state)
```

Comment:
Can this be removed?

```suggestion
```

It seems like random_state is always supplied. If so, I'd recommend making it a required argument.

Reply:
If those lines are removed, `random_state` is a numpy array and `pandas.core.common.random_state` raises as it's expecting integer, `np.random.RandomState`, or `None`

Reply:
Gotcha, thanks. I think pandas is being too strict there. I opened https://github.com/pandas-dev/pandas/issues/32503